### PR TITLE
Updated cpp-vgg13 per the new directory structure and added baselines' implementation

### PR DESCRIPTION
**Pull request by KumareshN93**
* Updated the cpp-vgg13 directory structure using cpp-vec-add as reference.
* Verified that the "make" and "make clean" work from both cpp-vgg13 directory and cpp-vgg13/PIM directory.
* Updated the script for baselines' (CPU and GPU) implementation. Verified that the script works for both CPU and GPU and GPU is much faster compared to CPU.
* Added a README file and slurm.sh file as well.

### CODE REVIEW

File: PIMbench/cpp-vgg13/README.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  
+## Compilation and Execution Instructions for Specific Variants
+
+### CPU Variant
+
+To run the script for the CPU variant, use command like the following example:
+
+```bash
+cd baselines
+python3 vgg13.py -s cpu -n 100
```
Comment (fasiddique) [inline PIMbench/cpp-vgg13/README.md:45]: Could you please elaborate what does `n` flag mean?

File: PIMbench/cpp-vgg13/baselines/vgg13.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    )
+
+    print("[INFO] Data loader finished")
+    return data_loader
+
+num_classes = 100  # CIFAR-100 has 100 classes
+if deviceStr == 'cpu':
+    batch_size = 128  # Adjust this based on your CPU's memory capacity
+if deviceStr == 'cuda':
+    batch_size = 2  # Adjust this based on your GPU's memory capacity
```
Comment (fasiddique) [inline PIMbench/cpp-vgg13/baselines/vgg13.py:56]: Is this memory in GB?

File: PIMbench/cpp-vgg13/baselines/vgg13.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        correct_top1 += (predicted_top1 == labels).sum().item()
+        correct_top5 += sum([1 if labels[i] in predicted_top5[i] else 0 for i in range(len(labels))])
+
+    top1_accuracy = 100 * correct_top1 / total
+    top5_accuracy = 100 * correct_top5 / total
+
+    print(f'Accuracy of the network on the test images (Top-1): {top1_accuracy:.2f} %')
+    print(f'Accuracy of the network on the test images (Top-5): {top5_accuracy:.2f} %')
+    print(f"Number of images: {args.num_images}")
+    print(f"Execution time per image: {(tm/args.num_images):.4f} seconds")
```
Comment (fasiddique) [inline PIMbench/cpp-vgg13/baselines/vgg13.py:102]: Can we print this in ms for consistency?

File: PIMbench/cpp-vgg13/baselines/vgg13.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    )
+
+    transform = transforms.Compose([
+            transforms.Resize((224, 224)),
+            transforms.ToTensor(),
+            normalize,
+    ])
+
+    if test:
+        dataset = datasets.CIFAR100(root=data_dir, train=False, download=True, transform=transform)
```
Comment (fasiddique) [inline PIMbench/cpp-vgg13/baselines/vgg13.py:40]: Can you please mention in the readme how much storage is needed for this?

File: PIMbench/cpp-vgg13/baselines/vgg13.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        dataset = Subset(dataset, range(num_images))
+    else:
+        dataset = datasets.CIFAR100(root=data_dir, train=True, download=True, transform=transform)
+
+    data_loader = torch.utils.data.DataLoader(
+        dataset, batch_size=batch_size, shuffle=shuffle, num_workers=4, pin_memory=True if device == 'cuda' else False
+    )
```
Comment (fasiddique) [inline PIMbench/cpp-vgg13/baselines/vgg13.py:47]: correct me if I am wrong, `num_workers` should be equal to hardware thread?

Comment (fasiddique) [reply PIMbench/cpp-vgg13/baselines/vgg13.py:47]: As we are doing inference do we need to shuffle?

File: PIMbench/cpp-vgg13/README.md (hunk 1 above)
Comment (KumareshN93) [reply PIMbench/cpp-vgg13/README.md:46]: Updated the README file. Kindly check.

File: PIMbench/cpp-vgg13/baselines/vgg13.py (hunk 2 above)
Comment (KumareshN93) [reply PIMbench/cpp-vgg13/baselines/vgg13.py]: The memory is typically in GB. Updated the comments in the vgg13.py file. Kindly check.

File: PIMbench/cpp-vgg13/baselines/vgg13.py (hunk 3 above)
Comment (KumareshN93) [reply PIMbench/cpp-vgg13/baselines/vgg13.py]: Updated the vgg13.py file. Kindly check.

File: PIMbench/cpp-vgg13/baselines/vgg13.py (hunk 4 above)
Comment (KumareshN93) [reply PIMbench/cpp-vgg13/baselines/vgg13.py]: Updated the comments in the README file. Kindly check.

File: PIMbench/cpp-vgg13/baselines/vgg13.py (hunk 5 above)
Comment (KumareshN93) [reply PIMbench/cpp-vgg13/baselines/vgg13.py]: * num_workers in this context refers to the number of subprocesses which can independently pre-process and load the data to speedup the process. Added relevant comments in the vgg13.py file. Kindly check.
* Yes, shuffle is not needed here. Updated the file accordingly. kindly check.

**Review by fasiddique — APPROVED** [review]
